### PR TITLE
Support TokenKeg as a known visualizer (instead of Unknown test)

### LINKED ISSUE #76: Solana - stopped showing all the instructions
### Project version

current

### What happened?

```
 grpcurl -plaintext -d '{"unsigned_payload": "AQAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAABAAEDGtcy7Vc3xB54TVH4H/JNV6GLORFZVW2eiFky1mqlJTJHohT28K37lWNJzHkspHGumVg0rwhDxT5hd/JUEGupaAbd9uHXZaGT2cvhRs7reawctIXtX1s3kTqM9YV+/wCpiz/aiPOGc/sEVBMImlZdQN5iFK0CVj9fTne9d3VuvB0BAgIBACMGAAF5QmVCZ074eW/VU/D+KlEJonY3BgtzkD1DFS0OaNFWDA==", "chain":"CHAIN_SOLANA"}'  localhost:44020 parser.ParserService/Parse
{
  "parsedTransaction": {
    "payload": {
      "signablePayload": "{\"Fields\":[{\"FallbackText\":\"Solana\",\"Label\":\"Network\",\"TextV2\":{\"Text\":\"Solana\"},\"Type\":\"text_v2\"},{\"FallbackText\":\"2omvYYYLCaCvtNWSUQK3XCJcAythXjrfp7S6ce7w6Jmo[SW], 5pdHyGbtCmZdJ7ye71nzeke8kcQ4ngJNPHqoDvE5L2WT[W], TokenkegQfeZyiNwAJbNbGKPFXCWuBvf9Ss623VQ5DA[R]\",\"Label\":\"Accounts\",\"PreviewLayout\":{\"Condensed\":{\"Fields\":[{\"FallbackText\":\"1 Signer\",\"Label\":\"Signers\",\"TextV2\":{\"Text\":\"1 Signer\"},\"Type\":\"text_v2\"},{\"FallbackText\":\"1 Writable\",\"Label\":\"Writable\",\"TextV2\":{\"Text\":\"1 Writable\"},\"Type\":\"text_v2\"},{\"FallbackText\":\"1 Read Only\",\"Label\":\"Read Only\",\"TextV2\":{\"Text\":\"1 Read Only\"},\"Type\":\"text_v2\"}]},\"Expanded\":{\"Fields\":[{\"FallbackText\":\"2omvYYYLCaCvtNWSUQK3XCJcAythXjrfp7S6ce7w6Jmo, Signer, Writable\",\"Label\":\"Account\",\"TextV2\":{\"Text\":\"2omvYYYLCaCvtNWSUQK3XCJcAythXjrfp7S6ce7w6Jmo, Signer, Writable\"},\"Type\":\"text_v2\"},{\"FallbackText\":\"5pdHyGbtCmZdJ7ye71nzeke8kcQ4ngJNPHqoDvE5L2WT, Writable\",\"Label\":\"Account\",\"TextV2\":{\"Text\":\"5pdHyGbtCmZdJ7ye71nzeke8kcQ4ngJNPHqoDvE5L2WT, Writable\"},\"Type\":\"text_v2\"},{\"FallbackText\":\"TokenkegQfeZyiNwAJbNbGKPFXCWuBvf9Ss623VQ5DA\",\"Label\":\"Account\",\"TextV2\":{\"Text\":\"TokenkegQfeZyiNwAJbNbGKPFXCWuBvf9Ss623VQ5DA\"},\"Type\":\"text_v2\"}]},\"Subtitle\":{\"Text\":\"3 accounts\"},\"Title\":{\"Text\":\"Accounts\"}},\"Type\":\"preview_layout\"}],\"PayloadType\":\"SolanaTx\",\"Title\":\"Solana Transaction\",\"Version\":\"0\"}"
    },
    "signature": {
      "scheme": "SIGNATURE_SCHEME_TURNKEY_P256_EPHEMERAL_KEY",
      "publicKey": "04716208f9a297aebde668b949fbabd23a2d92c6d3017abf56af9b2e27fac889ec3f99f033e82872f323a7ac124ffc4dfe19bf67fd93cc8070aba224525564aaef04fdb364b9c337303db3df2e39efecd7667cb00bfa197e2cc0a850e6846e5d3f715d55371d0f64cc5c46e28ff8bb00bd403508e4120c337b257b6b65baed68bd57",
      "message": "4cab20a2594abc50dc963839b169786bc23af110cb9e8090c1f414d6cb358be7",
      "signature": "150e0e3483f4fd8ee4301367d1b8125763b22f6f3feede97cb301848f644b885979d80ce00f72d2861c85744faff89e00c56048289f0331045b5760357c42a04"
    }
  }
```
this is only showing account keys and no longer list of instructions.. but when you decode with solana-parser it shows

```
cargo run parse --transaction "0100000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000010001031ad732ed5737c41e784d51f81ff
24d57a18b391159556d9e885932d66aa5253247a214f6f0adfb956349cc792ca471ae995834af0843c53e6177f254106ba96806ddf6e1d765a193d9cbe146ceeb79ac1cb485ed5f5b37913a8cf5857eff00a98b3fda88f38673fb045413089a565d40d
e6214ad02563f5f4e77bd77756ebc1d01020201002306000179426542674ef8796fd553f0fe2a5109a27637060b73903d43152d0e68d1560c"
    Finished `dev` profile [unoptimized + debuginfo] target(s) in 0.12s
     Running `target/debug/solana_parser parse --transaction 0100000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000010001031ad732ed5737c41e784d51f81ff24d57a18b391159556d9e885932d66aa5253247a214f6f0adfb956349cc792ca471ae995834af0843c53e6177f254106ba96806ddf6e1d765a193d9cbe146ceeb79ac1cb485ed5f5b37913a8cf5857eff00a98b3fda88f38673fb045413089a565d40de6214ad02563f5f4e77bd77756ebc1d01020201002306000179426542674ef8796fd553f0fe2a5109a27637060b73903d43152d0e68d1560c`
Solana Parsed Transaction Payload:
  Unsigned Payload: 0100000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000010001031ad732ed5737c41e784d51f81ff24d57a18b391159556d9e885932d66aa5253247a214f6f0adfb956349cc792ca471ae995834af0843c53e6177f254106ba96806ddf6e1d765a193d9cbe146ceeb79ac1cb485ed5f5b37913a8cf5857eff00a98b3fda88f38673fb045413089a565d40de6214ad02563f5f4e77bd77756ebc1d01020201002306000179426542674ef8796fd553f0fe2a5109a27637060b73903d43152d0e68d1560c
  Transaction Metadata:
    Signatures: ["00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000"]
    Account Keys: ["2omvYYYLCaCvtNWSUQK3XCJcAythXjrfp7S6ce7w6Jmo", "5pdHyGbtCmZdJ7ye71nzeke8kcQ4ngJNPHqoDvE5L2WT", "TokenkegQfeZyiNwAJbNbGKPFXCWuBvf9Ss623VQ5DA"]
    Program Keys: ["TokenkegQfeZyiNwAJbNbGKPFXCWuBvf9Ss623VQ5DA"]
    Recent Blockhash: ANa9GK3wsd73HP8pw9pUXjBe7E8Z5ZsDeZYGuqeZWF3A
    Instructions:
      Instruction 1:
        Program Key: TokenkegQfeZyiNwAJbNbGKPFXCWuBvf9Ss623VQ5DA
        Accounts: [SolanaAccount { account_key: "5pdHyGbtCmZdJ7ye71nzeke8kcQ4ngJNPHqoDvE5L2WT", signer: false, writable: true }, SolanaAccount { account_key: "2omvYYYLCaCvtNWSUQK3XCJcAythXjrfp7S6ce7w6Jmo", signer: true, writable: true }]
        Instruction Data (hex): 06000179426542674ef8796fd553f0fe2a5109a27637060b73903d43152d0e68d1560c
        Address Table Lookups: []
    Transfers:
    SPL Transfers:
    Address Table Lookups: []
```

### Minimal reproduction steps

construct a transaction in Solana that has a program we don't already decode like TokenKeg

### PR DESCRIPTION
**Pull request by prasanna-anchorage**
## Support TokenKeg (SPL Token) program

  Resolves #76

  ### Summary
  Adds dedicated visualizer for the TokenKeg program (`TokenkegQfeZyiNwAJbNbGKPFXCWuBvf9Ss623VQ5DA`) to display meaningful SPL
  token instruction details instead of showing as "unknown program". This enables SPL token minters to understand what
  operations are being performed in their transactions.

  ### Changes
  - **New SPL Token Preset** (`src/presets/spl_token/`)
    - Implements full instruction parsing using `TokenInstruction::unpack()` from `spl-token` crate
    - Supports all major instructions: MintTo, MintToChecked, Transfer, TransferChecked, Burn, InitializeMint, etc.
    - Special handling for mint operations with detailed amount and decimals display
    - Follows existing visualizer patterns for consistency

  - **Test Coverage**
    - Renamed `test_unknown_program_tokenkeg` → `test_spl_token_tokenkeg_recognition` to verify TokenKeg is properly recognized
  as SPL Token
    - Added `test_unknown_program_fallback` with fake program ID pattern (FAKEPROGRAM!) to validate unknown program visualizer
  still works
    - All 28 unit tests pass

  - **Build Integration**
    - Registered in `presets/mod.rs` for auto-discovery by build.rs
    - Placed before `UnknownProgramVisualizer` in visualizer chain

  ### Example Output
  Before (Unknown Program):
  Program ID: TokenkegQfeZyiNwAJbNbGKPFXCWuBvf9Ss623VQ5DA
  Instruction Data: 0601014f42...

  After (SPL Token):
  Instruction: Mint To
  Amount: 1000000
  Decimals: 6
  Program: SPL Token

  ### Testing
  ```bash
  cargo test --lib  # All 28 tests pass
  cargo clippy -- -D warnings  # No warnings
  cargo fmt --check  # Properly formatted

### CODE REVIEW

File: src/chain_parsers/visualsign-solana/src/presets/spl_token/tests.rs [hunk 1]
```diff
@@ -0,0 +1,1243 @@
...
+
+/// Test case for simple instructions (no parameters)
+struct SimpleTestCase {
+    name: &'static str,
+    expected_name: &'static str,
+    builder: fn(&Pubkey, &Pubkey, &Pubkey) -> solana_sdk::instruction::Instruction,
+    variant_check: fn(&TokenInstruction) -> bool,
+}
+
+fn run_amount_test(test: &AmountTestCase) {
```
Comment (diego-rivas-anchor) [inline src/chain_parsers/visualsign-solana/src/presets/spl_token/tests.rs:62]: Can we add test cases for malformed/empty/truncated instruction data? every test is building well-formed data

File: src/chain_parsers/visualsign-solana/src/presets/swig_wallet/mod.rs [hunk 2]
```diff
@@ -862,7 +862,11 @@ fn visualize_inner_instruction(instruction: Instruction) -> Option<String> {
     visualize_with_any(&visualizer_refs, &context)
         .and_then(|result| result.ok())
         .and_then(|viz_result| match viz_result.kind {
-            VisualizerKind::Payments("UnknownProgram") => None,
+            // Skip catch-all and SPL Token; both produce summaries that are less informative
+            // than swig's own format_token_instruction_summary for inner instructions.
+            VisualizerKind::Payments("UnknownProgram") | VisualizerKind::Payments("SplToken") => {
+                None
+            }
```
Comment (diego-rivas-anchor) [inline src/chain_parsers/visualsign-solana/src/presets/swig_wallet/mod.rs:920]: Is this change related to TokenKeg support?

Comment (prasanna-anchorage) [reply src/chain_parsers/visualsign-solana/src/presets/swig_wallet/mod.rs:920]: Yes, directly. Before this branch, `visualize_with_any` had no `SplToken` visualizer registered, so SPL Token inner instructions never matched there and always fell through to swig's own `format_token_instruction_summary` (which produces e.g. "Transfer 1.000000 USDC" using token metadata). This PR registers `SplTokenVisualizer`, which would otherwise start matching at `visualize_with_any` and return a less specific summary like "Mint To: 1000" for inner instructions inside swig. Adding `SplToken` to the skip arm preserves swig's existing inner-instruction rendering. `UnknownProgram` was already skipped for the same reason.

File: src/chain_parsers/visualsign-solana/src/presets/spl_token/tests.rs (hunk 1 above)
Comment (prasanna-anchorage) [reply src/chain_parsers/visualsign-solana/src/presets/spl_token/tests.rs:62]: Good catch — added four tests in the next commit:

- `test_unpack_rejects_empty_data` — `TokenInstruction::unpack(&[])` returns `Err`
- `test_unpack_rejects_truncated_transfer_data` — Transfer tag with no amount bytes returns `Err`
- `test_unpack_rejects_unknown_opcode` — tag `0xFF` returns `Err`
- `test_visualizer_returns_decode_error_for_malformed_data` — end-to-end: `SplTokenVisualizer::visualize_tx_commands` with empty data returns `Err(VisualSignError::DecodeError(_))` rather than panicking. This is the real regression guard for malformed wire input.

Comment (shahan-khatchadourian-anchorage) [reply src/chain_parsers/visualsign-solana/src/presets/spl_token/tests.rs:62]: Flagging before this thread is resolved: the malformed/empty/truncated coverage added here is good, but `test_visualizer_returns_decode_error_for_malformed_data` asserts the *visualizer* returns `Err(DecodeError)`, which conflicts with the preset's intended contract (PRS-464 design decision #2: graceful fallback, never `Err`). A visualizer `Err` aborts the whole transaction in the non-diagnostics build — the blocker noted on this PR. The three `unpack`-level `Err` tests are correct (they exercise the spl-token SDK). The end-to-end one should be flipped to assert a graceful `Ok` raw/placeholder layout, and a v0 + address-lookup-table (unresolved-accounts) case added as the other half of the malformed-input coverage. Leaving this thread open until that lands.

**Review by shahan-khatchadourian-anchorage — CHANGES_REQUESTED** [review]
**Blocker:** v0 SPL instructions with lookup-table accounts hit `Unresolved`, which `build_instruction_shim` turns into an error that erases the whole tx in production (dispatch does not fall back to `unknown_program`). Degrade to `unresolved(N)` placeholders instead of erroring + add a v0+ALT test. Rest LGTM; no security findings.

Related: the malformed-data test thread ([discussion_r3363242170](https://github.com/anchorageoss/visualsign-parser/pull/81#discussion_r3363242170)) stays open under this — `test_visualizer_returns_decode_error_for_malformed_data` currently asserts `Err` and should be flipped to assert graceful `Ok` rendering, with a v0+ALT case added.

File: src/chain_parsers/visualsign-solana/src/presets/spl_token/tests.rs (hunk 1 above)
Comment (prasanna-anchorage) [reply src/chain_parsers/visualsign-solana/src/presets/spl_token/tests.rs:62]: Both halves landed in 2cfeab8.

Root cause was wider than the test: `build_instruction_shim` returned `Err` for *any* unresolved account index, so a v0 + lookup-table SPL instruction (program_id resolves to TokenKeg, so spl_token handles it, but token accounts come from the ALT) erased the whole tx in the non-diagnostics dispatch — same failure mode as the malformed-bytes path.

Changes:
- Replaced the fallible shim with an infallible `InstructionView` that resolves program/account indices to display strings and renders unresolved (v0+ALT) indices as `unresolved(N)` placeholders, matching the catch-all "partial rendering" contract documented on `VisualizerContext` (the `unknown_program` convention).
- On `TokenInstruction::unpack` failure, degrade to a raw program_id + data layout (`Ok`) instead of `DecodeError`.

Tests:
- `test_visualizer_returns_decode_error_for_malformed_data` -> `test_visualizer_renders_graceful_fallback_for_malformed_data`, now asserting a graceful `Ok` raw layout.
- Added `test_v0_unresolved_accounts_render_placeholder_not_err`: a TokenKeg MintTo whose authority index points into an unresolved ALT slot, asserting an `unresolved(50)` placeholder rather than an aborted transaction.

The three `unpack`-level `Err` tests are unchanged (they exercise the spl-token SDK directly). Full solana suite passes under default and `diagnostics` features; clippy `-D warnings` clean.

**Comment by prasanna-anchorage** [comment]
Rebased onto current `main` (`58bbde9`) and ready for re-review.

- Blocker addressed: spl_token no longer aborts the transaction on v0 + address-lookup-table accounts (renders `unresolved(N)` placeholders) or malformed bytes (graceful raw layout). Details in the [resolved thread](https://github.com/anchorageoss/visualsign-parser/pull/81#discussion_r3376035813).
- Rebase: only `.gitignore` conflicted (kept both blocks); `core/visualsign.rs` and `swig_wallet/mod.rs` auto-merged.
- Green against the new base: 240 lib tests + all integration suites pass; `clippy -D warnings` and `fmt --check` clean under both default and `diagnostics` features.
- Follow-up for the same `resolve_accounts()?` pattern across the other IDL presets: #362.

**Review by shahan-khatchadourian-anchorage — APPROVED** [review]
All changes that I requested are addressed. lgtm.
